company/validation.go

```diff
@@ -1,17 +1,48 @@
 package company
 
 import (
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-
-	pb "github.com/dinhtp/lets-go-pbtype/company"
+    "strings"
 
+    pb "github.com/dinhtp/lets-go-pbtype/company"
+    "google.golang.org/grpc/codes"
+    "google.golang.org/grpc/status"
 )
 
 func validateOne(r *pb.OneCompanyRequest) error {
-	if "" == r.GetId() {
-		return status.Error(codes.InvalidArgument, "company id is required")
-	}
+    if "" == r.GetId() {
+        return status.Error(codes.InvalidArgument, "company id is required")
+    }
+    return nil
+}
+
+func validateCreate(r *pb.Company) error {
+    if r.GetName() == "" || r.GetPhone() == "" || r.GetEmail() == "" || r.GetAddress() == "" || r.GetTaxNumber() == "" {
+        return status.Error(codes.InvalidArgument, "Attributed is required")
+    }
+    return nil
+}
+
+func validateUpdate(r *pb.Company) error {
+    if "" == r.GetId() {
+        return status.Error(codes.InvalidArgument, "company id is required")
+    }
+    return validateCreate(r)
+}
+
+func validateList(r *pb.ListCompanyRequest) error {
+    var page = r.GetPage()
+    var limit = r.GetLimit()
+    var field = strings.Split(r.GetSearchField(), ",")
+
+    if (page <= 0) || (limit <= 0) {
+        return status.Error(codes.InvalidArgument, "Invalid Attribute")
```

Comment:
Tách từng điều kiện ra, vì cần cả page và limit mới phân trang được
khi ko có page thì ghi hẳn lỗi là page is required hoặc page is invalid
tương tự với limit